cmd/collector/app/span_processor_test.go

```diff
@@ -807,3 +822,204 @@ func TestSpanProcessorWithOnDroppedSpanOption(t *testing.T) {
 	require.EqualError(t, err, processor.ErrBusy.Error())
 	assert.Equal(t, []string{"op3"}, droppedOperations)
 }
+
+func optionsWithPorts(portHttp string, portGrpc string) *cFlags.CollectorOptions {
+	opts := &cFlags.CollectorOptions{
+		OTLP: struct {
+			Enabled bool
+			GRPC    configgrpc.ServerConfig
+			HTTP    confighttp.ServerConfig
+		}{
+			Enabled: true,
+			HTTP: confighttp.ServerConfig{
+				Endpoint:        portHttp,
+				IncludeMetadata: true,
+			},
+			GRPC: configgrpc.ServerConfig{
+				NetAddr: confignet.AddrConfig{
+					Endpoint:  portGrpc,
+					Transport: confignet.TransportTypeTCP,
+				},
+			},
+		},
+	}
+	return opts
+}
+
+func TestOTLPReceiverWithV2Storage(t *testing.T) {
+	// Send trace via HTTP
+	t.Run("Send trace data using HTTP connection", func(t *testing.T) {
+		mockWriter := mocks.NewWriter(t)
+
+		spanProcessor, err := NewSpanProcessor(
+			mockWriter,
+			nil,
+			Options.NumWorkers(1),
+			Options.QueueSize(1),
+			Options.ReportBusy(true),
+		)
+		require.NoError(t, err)
+		defer spanProcessor.Close()
+		logger := zaptest.NewLogger(t)
+
+		portHttp := "4317"
+		portGrpc := "4318"
+		var receivedTraces atomic.Pointer[ptrace.Traces]
+		var receivedCtx atomic.Pointer[context.Context]
+		mockWriter.On("WriteTraces", mock.Anything, mock.Anything).
+			Run(func(args mock.Arguments) {
+				storeContext := args.Get(0).(context.Context)
+				storeTrace := args.Get(1).(ptrace.Traces)
+				receivedTraces.Store(&storeTrace)
+				receivedCtx.Store(&storeContext)
+			}).Return(nil)
+		// Can't send tenancy headers with http request to OTLP receiver
+		tenancyMgr := tenancy.NewManager(&tenancy.Options{
+			Enabled: true,
+			Header:  "x-tenant",
+			Tenants: []string{"test-tenant"},
+		})
+
+		rec, err := handler.StartOTLPReceiver(
+			optionsWithPorts(fmt.Sprintf("localhost:%v", portHttp), fmt.Sprintf("localhost:%v", portGrpc)),
+			logger,
+			spanProcessor,
+			tenancyMgr,
+		)
+		require.NoError(t, err)
+		ctx := context.Background()
+		defer rec.Shutdown(ctx)
```

Comment:
from L852 you are repeating a lot of code for each test. And each time you are starting both servers, so all that repetition is unnecessary. You can create a helper function that will initialize the receiver.